azext_iot/operations/events3/_events.py

```diff
@@ -401,3 +355,51 @@ def handle_msg(msg):
 
 def _get_container_id():
     return "{}/{}".format(USER_AGENT, str(uuid4()))
+
+
+def _output_msg_kpi(
+    msg,
+    device_id,
+    devices,
+    pnp_context,
+    interface_name,
+    content_type,
+    properties,
+    output,
+    validate_messages,
+    simulate_errors,
+):
+    parser = Event3Parser()
```

Comment:
Can we refactor the parser initialization so we don't instantiate a new Event3Parser for every message?

Reply:
How strongly do you feel about this? I think this will be tricky due to thread safety issues since these messages are being parsed async.

The issue is that the error/warning arrays are encapsulated within the class so that the caller doesn't have to worry about implementation details. Since we need to clear these each time a new message comes in, we risk deleting the errors/warnings from another message.

I also believe the extra memory used by this shouldn't be a concern. And since we don't really hold on to the EventParser object, it should get garbage collected once the function returns

Reply:
nevermind, on second look validation is done per device event, so this is OK.

Reply:
#wontfix